src/platform_impl/macos/window.rs

```diff
@@ -247,6 +249,29 @@ fn create_window(
       if !pl_attrs.has_shadow {
         ns_window.setHasShadow_(NO);
       }
+
+      // titlebar_style_hidden
+      {
+        use cocoa::appkit::NSWindowTitleVisibility;
+        let mut style_mask = ns_window.styleMask();
+        style_mask.set(
+          NSWindowStyleMask::NSFullSizeContentViewWindowMask,
+          pl_attrs.titlebar_style_hidden,
+        );
+        ns_window.setStyleMask_(style_mask);
+
+        ns_window.setTitleVisibility_(if pl_attrs.titlebar_style_hidden {
+          NSWindowTitleVisibility::NSWindowTitleHidden
+        } else {
+          NSWindowTitleVisibility::NSWindowTitleVisible
+        });
+        ns_window.setTitlebarAppearsTransparent_(if pl_attrs.titlebar_style_hidden {
+          cocoa::base::YES
+        } else {
+          cocoa::base::NO
+        });
+      }
```

Comment:
You can move the if statement to the top of this block and get rid of the inline checks. All the else branches are just the default values.

Reply:
Good call, that's much better!